compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -530,12 +530,33 @@ impl<'a, 'tcx> Visitor<'tcx> for BoundVarContext<'a, 'tcx> {
                 // These sorts of items have no lifetime parameters at all.
                 intravisit::walk_item(self, item);
             }
-            hir::ItemKind::Static(..) | hir::ItemKind::Const(..) => {
+            hir::ItemKind::Static(..) => {
                 // No lifetime parameters, but implied 'static.
                 self.with(Scope::Elision { s: self.scope }, |this| {
                     intravisit::walk_item(this, item)
                 });
             }
+            // FIXME(generic_const_items): Somehow merge this branch with the `TypeAlias` one below?
+            hir::ItemKind::Const(_, generics, _) => {
+                self.with(Scope::Elision { s: self.scope }, |this| {
```

Comment:
This scope should not be necessary. `Elision` here only acts on object lifetime default. The current scope is already `Root`, which has the same effect.
=> this can be merged with the general arm below.